### PR TITLE
TPG and Pedestals tools, DBv2 updates

### PR DESCRIPTION
**Pull request by depasse**
CondDBCommon to suppress
Ecal TPGPedestals and Pedestals tools

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by davidlt** [comment]
Very quick look and does not seem that `sprintf` is used here correctly.

**Comment by mmusich** [comment]
please test

File: CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc [hunk 1]
```diff
@@ -23,9 +23,15 @@
 #include "CondFormats/EcalObjects/interface/EcalTPGPedestals.h"
 #include "CondFormats/DataRecord/interface/EcalTPGPedestalsRcd.h"
 
+//modif-alex-27-july-2015
+#include "CalibCalorimetry/EcalLaserCorrection/interface/EcalLaserDbService.h"
+#include "CalibCalorimetry/EcalLaserCorrection/interface/EcalLaserDbRecord.h"
+
 #include "SimCalorimetry/EcalSimAlgos/interface/EcalSimParameterMap.h"
+#if (CMSSW_VERSION>=340)
 #include "SimCalorimetry/EcalSimAlgos/interface/EBShape.h"
```
Comment (mmusich) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc]: @depasse, can you clarify why do you need this protection? is this still running in a 3.4.X version somewhere (???)

File: CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc [hunk 2]
```diff
@@ -317,6 +327,33 @@ EcalTPGParamBuilder::EcalTPGParamBuilder(edm::ParameterSet const& pSet)
 
   std::cout << "INFO: DONE reading timing files for EB and EE" << std::endl;
 
+  //modif-alex-30/01/2012
+  if(useTransparencyCorr_){
+    std::cout << "INFO: READING transparency correction files" << std::endl;
+    std::ifstream transparency(Transparency_Corr_.c_str());
```
Comment (mmusich) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc]: @depasse please replace `cout` with `LogInfo`

**Comment by mmusich** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by mmusich** [comment]
+1

**Comment by rekovic** [comment]
+1

File: CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc [hunk 3]
```diff
@@ -376,6 +396,38 @@ void EcalTPGParamBuilder::analyze(const edm::Event& evt, const edm::EventSetup&
...
+  const EcalLaserAlphaMap& laserAlphaMap = handle.product()->getMap(); // map of apdpns
+
+  //modif-alex-27-july-2015-beg
+  // use alpha to check
+  int cnt =0;
+  for(int ieta=-EBDetId::MAX_IETA; ieta<=EBDetId::MAX_IETA; ++ieta) {
+    if(ieta==0) continue;
+    for(int iphi=EBDetId::MIN_IPHI; iphi<=EBDetId::MAX_IPHI; ++iphi) {
+      EBDetId ebdetid(ieta,iphi);
+      EcalLaserAlphaMap::const_iterator italpha = laserAlphaMap.find( ebdetid );
```
Comment (davidlange6) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc]: @depasse - it looks like this find inside a double loop is better implemented as a loop over laserAlphaMap (you can fill a map of ieta iphi if the check for missing laser conditions is important).

File: CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.h [hunk 4]
```diff
@@ -1,6 +1,12 @@
 #ifndef ECALTPGPARAMBUILDER_H
 #define ECALTPGPARAMBUILDER_H
 
+//Author: Pascal Paganini - LLR
+//Date: 2006/07/10 15:58:06 $
+
+#define CMSSW_VERSION 340
```
Comment (davidlange6) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.h]: why is this needed?

Comment (depasse) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.h]: It is historical.  I'm not the author of this pgm.  I don't know if somewhere a version < 3_4_0 uses this condition.  Removing this definition and the tests after, works, from a compilation point a view.

File: CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc (hunk 3 above)
Comment (depasse) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.cc]: We will work for a better version.

File: CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.h (hunk 4 above)
Comment (depasse) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.h]: Sorry, compilation does NOT work any more, removing this test.  Many references for EcalShape in .cc...I don't know if it is used elsewhere.

Comment (davidlange6) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.h]: your PR seems to have added it , so indeed there are likely places in the code with this 6 years obsolete stuff - however we don't need to add new if statements as done in this pr

Comment (depasse) [inline CalibCalorimetry/EcalTPGTools/plugins/EcalTPGParamBuilder.h]: Added ?? No, it was the original version since many years.  As there were multiple commits in this PR, maybe I removed it and reused an older version after..
In any case, I will resubmit a new .h and .cc  without this if 340 (hoping there will be no consequence but I don't think so)  and resolve the double for loop in the .cc.
Thanks.

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by ggovi** [comment]
+1

**Comment by mmusich** [comment]
@mulhearn @rekovic could you please re-check and sign-off? Thanks in advance,

**Comment by mmusich** [comment]
Hi Pierre, 
is the last commit intended as final? Can we re-trigger the tests?

**Comment by depasse** [comment]
Hi Marco,

yes,

as edm:LogInfo was not correct (but without consequences) and there was still a trigger waiting, I heve relaunched the modifications.  Thanks.

Pierre
________________________________

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
+1

**Comment by ggovi** [comment]
+1
